subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c

```diff
@@ -131,29 +131,68 @@ static const char *const agps_req_type_strings[] = {
 	[NRF_CLOUD_REST_AGPS_REQ_CUSTOM]	= AGPS_REQ_TYPE_STR_CUSTOM,
 };
 
+/* Generate an authorization header value string in the form:
+ * "Authorization: Bearer JWT \r\n"
+ */
 static int generate_auth_header(const char *const tok, char **auth_hdr_out)
 {
-	if (!tok || !auth_hdr_out) {
```

Comment:
this check `!tok && !IS_ENABLED(CONFIG_NRF_CLOUD_REST_AUTOGEN_JWT)` can be done here to prevent and unnecessary malloc.